internal/srv/handlers.go

```diff
@@ -42,7 +46,7 @@ func (s *Server) processEvent(messages <-chan *message.Message) {
 				}
 			}
 
-			if lb.lbType != typeNoLB {
+			if lb != nil && lb.lbType != typeNoLB {
```

Comment:
Should we decouple the nil check and fail faster? If LB data isn't found from one of the above, we probably want to fail faster and provide a more unique error message.

Reply:
we've got the upstream error logged. We could put in another `continue` there on line 46. I think the extra guard to check the lb for nil is a nice extra safety measure